cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -146,6 +137,20 @@ protected boolean removeEldestEntry(Map.Entry<UUID, UserTaskInfo> eldest) {
     this(sessionExpiryMs, maxActiveUserTasks, completedUserTaskRetentionTimeMs, maxCachedCompletedUserTasks, time, new UUIDGenerator());
   }
 
+  private void initUuidToCompletedUserTaskInfoMap(int maxCachedCompletedMonitorUserTasks,
+                                                  int maxCachedCompletedAdminUserTasks) {
+    for (RetentionType retentionType : RetentionType.cachedValues()) {
+      int maxCachedCompletedUserTasks = (retentionType == ADMIN_CRUISE_CONTROL || retentionType == ADMIN_KAFKA) ?
```

Comment:
I change this to a `switch` statement so that anytime the assumption does not hold and we forget to change the logic here, an `Exception` will be thrown.